test/parallel/test-util-styletext.js

```diff
@@ -46,6 +46,20 @@ assert.strictEqual(
   '\u001b[1m\u001b[31mtest\u001b[39m\u001b[22m',
 );
 
+assert.strictEqual(
+  util.styleText('red',
+                 'A' + util.styleText('blue', 'B', { validateStream: false }) + 'C',
+                 { validateStream: false }),
+  '\u001b[31mA\u001b[34mB\u001b[31mC\u001b[39m'
+);
+
+assert.strictEqual(
+  util.styleText('red',
+                 'A' + util.styleText(['bgRed', 'blue'], 'B', { validateStream: false }) +
+    'C', { validateStream: false }),
+  '\x1B[31mA\x1B[41m\x1B[34mB\x1B[31m\x1B[49mC\x1B[39m'
+);
```

Comment:
we should add a test for deep nesting too, to ensure the algo works more than depth-1

`<blue>text <red>red text <green>green text</green> red text</red> blue text</blue>`